extensions/ql-vscode/src/view/model-editor/ModelKindDropdown.tsx

```diff
@@ -63,7 +63,16 @@ export const ModelKindDropdown = ({
 
   useEffect(() => {
     const value = modeledMethod?.kind;
```

Comment:
I find the logic in this method a little confusing because of multiple `if` statements in sequence where some return early and some don't. I think we could also perhaps make `value` never be undefined and that could simplify the logic.

What do you think of:
```typescript
const value = modeledMethod?.kind ?? "";

if (kinds.length === 0 && value !== "") {
  onChangeKind("");
} else if (kinds.length > 0 && !kinds.includes(value)) {
  onChangeKind(kinds[0]);
}
```